homeassistant/components/switchbot_cloud/entity.py

```diff
@@ -0,0 +1,49 @@
+"""Base class for SwitchBot via API entities."""
+import logging
+from typing import Any
+
+from switchbot_api import Device, Remote, SwitchBotAPI
+
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SwitchBotCloudEntity(CoordinatorEntity):
+    """Representation of a SwitchBot Cloud entity."""
+
+    _api: SwitchBotAPI
+    _switchbot_state: dict[str, Any] | None = None
+
+    def __init__(
+        self,
+        api: SwitchBotAPI,
+        device: Device | Remote,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator=coordinator)
+        self._api = api
+        self._attr_unique_id = device.device_id
+        self._attr_name = device.device_name
```

Comment:
New integrations have to use `has_entity_name = True`. For more info check https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations

Reply:
In your case, add `_attr_has_entity_name = True` to this base entity and since you current use the device name as name, set `_attr_name = None` in the switch.

Reply:
@joostlek Thanks for pointing this issue, I totally missed it